src/rules.c

```diff
@@ -133,6 +132,34 @@ SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, IndexSpec *spec, QueryError
   return NULL;
 }
 
+void RSExpr_GetProperties(RSExpr *e, char ***props);
+
+void SchemaRule_FilterFields(SchemaRule *rule) {
+  // if filter exist, check which fields should be loaded
+  char **properties = array_new(char *, 8);
+  IndexSpec *spec = rule->spec;
+  RSExpr_GetProperties(rule->filter_exp, &properties);
+  int propLen = array_len(properties);
+  if (array_len(properties) > 0) {
+    rule->filter_fields = properties;
+    rule->filter_fields_index = rm_calloc(propLen, sizeof(int));
+    for (int i = 0; i < propLen; ++i) {
+      for (int j = 0; j < spec->numFields; ++j) {
+        FieldSpec *fs = spec->fields + j;
+        if (!strcmp(properties[i], fs->name) || !strcmp(properties[i], fs->path)) {
+          rule->filter_fields_index[i] = j;
+          break;
+        }
+        // no match was found
+        rule->filter_fields_index[i] = -1;
```

Comment:
What does it mean if there is no match? Why we even need to search the properties in the index definition? Maybe I am missing something but its not clear to me.

Reply:
We need to search the definition because we may have field->path and field->name that are not the same. Or we can have a filter on a field that is not in the schema.